initgeneration/initgeneration_test.go

```diff
@@ -0,0 +1,328 @@
+package initgeneration
+
+import (
+	"context"
+	_ "embed"
+	"testing"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/logger"
+	"github.com/Dynatrace/dynatrace-operator/mapper"
+	"github.com/Dynatrace/dynatrace-operator/scheme/fake"
+	"github.com/Dynatrace/dynatrace-operator/webhook"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+//go:embed init.sh.test-sample
+var scriptSample string
+
+var (
+	operatorNamespace        = "dynatrace"
+	testNamespaceName        = "namespace"
+	testOtherNamespaceName   = "other-namespace"
+	testDynakubeComplexName  = "dynakubeComplex"
+	testDynakubeSimpleName   = "dynakubeSimple"
+	testTokensName           = "kitchen-sink"
+	testApiUrl               = "https://test-url/api"
+	testProxy                = "testproxy.com"
+	testtrustCAsCM           = "testtrustedCAsConfigMap"
+	testCAValue              = "somecertificate"
+	testTenantUUID           = "abc12345"
+	kubesystemNamespace      = "kube-system"
+	kubesystemUID            = types.UID("42")
+	testNode1Name            = "node1"
+	testNode2Name            = "node2"
+	testNodeWithSelectorName = "nodeWselector"
+	testSelectorLabels       = map[string]string{"test": "label"}
+
+	testDynakubeComplex = &dynatracev1alpha1.DynaKube{
```

Comment:
different test setup compared to other places. 
could be changed to use setup functions